File: app/src/main/java/com/perol/asdpl/pixivez/services/AppApiPixivService.java
package com.perol.asdpl.pixivez.services;


import com.perol.asdpl.pixivez.responses.AboutPictureResponse;
import com.perol.asdpl.pixivez.responses.BookMarkDetailResponse;
import com.perol.asdpl.pixivez.responses.BookMarkTagsResponse;
import com.perol.asdpl.pixivez.responses.FollowResponse;
import com.perol.asdpl.pixivez.responses.IllustCommentsResponse;
import com.perol.asdpl.pixivez.responses.IllustDetailResponse;
import com.perol.asdpl.pixivez.responses.IllustRankingResponse;
import com.perol.asdpl.pixivez.responses.IllustfollowResponse;
import com.perol.asdpl.pixivez.responses.PixivResponse;
import com.perol.asdpl.pixivez.responses.RecommendResponse;
import com.perol.asdpl.pixivez.responses.SearchIllustResponse;
import com.perol.asdpl.pixivez.responses.SearchUserResponse;
import com.perol.asdpl.pixivez.responses.SpotlightResponse;
import com.perol.asdpl.pixivez.responses.TrendingtagResponse;
import com.perol.asdpl.pixivez.responses.UgoiraMetadataResponse;
import com.perol.asdpl.pixivez.responses.UserDetailResponse;
import com.perol.asdpl.pixivez.responses.UserIllustsResponse;
import com.perol.asdpl.pixivez.responses.WalkthroughResponse;

import java.util.List;

import io.reactivex.Observable;
import kotlinx.coroutines.Deferred;
import okhttp3.MultipartBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.Query;
import retrofit2.http.Streaming;
import retrofit2.http.Url;

/**
 * Created by asdpl on 2018/2/10.
 */

public interface AppApiPixivService {
    //    @FormUrlEncoded
//    @POST("/v1/mute/edit")
//    public abstract l<PixivResponse> postMuteSetting(@Header("Authorization") String paramString, @Field("add_user_ids[]") List<Long> paramList1, @Field("delete_user_ids[]") List<Long> paramList2, @Field("add_tags[]") List<String> paramList3, @Field("delete_tags[]") List<String> paramList4);
    @GET("/v2/illust/bookmark/detail")
    Observable<BookMarkDetailResponse> getLikeIllustDetail(@Header("Authorization") String paramString, @Query("illust_id") long paramLong);

    @GET("v1/user/bookmark-tags/illust")
    Observable<BookMarkTagsResponse> getIllustBookmarkTags(@Header("Authorization") String paramString1, @Query("user_id") long paramLong, @Query("restrict") String paramString2);

    @GET("v1/walkthrough/illusts")
    Observable<WalkthroughResponse> getWalkthroughIllusts();

    @GET
    Observable<BookMarkTagsResponse> getNexttags(@Header("Authorization") String paramString1, @Url String paramString2);

    @GET("/v1/spotlight/articles?filter=for_ios")
    Observable<SpotlightResponse> getPixivisionArticles(@Header("Authorization") String paramString1, @Query("category") String paramString2);

    @GET("/v1/user/recommended?filter=for_android")
    Observable<SearchUserResponse> getUserRecommended(@Header("Authorization") String paramString);

    @GET
    Observable<SpotlightResponse> getNextPixivisionArticles(@Header("Authorization") String paramString1, @Url String paramString2);

    @Streaming
    @GET
    Observable<ResponseBody> getGIFFile(@Url String fileUrl);

    @Multipart
    @POST("/v1/user/profile/edit")
    Observable<ResponseBody> postUserProfileEdit(@Header("Authorization") String paramString, @Part MultipartBody.Part paramRequestBody);

    @GET("/v1/ugoira/metadata")
    Observable<UgoiraMetadataResponse> getUgoiraMetadata(@Header("Authorization") String paramString, @Query("illust_id") long paramLong);

    @GET("/v1/user/browsing-history/illusts")
    Observable<UserIllustsResponse> getIllustBrowsingHistory(@Header("Authorization") String paramString);

    @GET("/v1/user/bookmarks/illust")
    Observable<UserIllustsResponse> getLikeIllust(@Header("Authorization") String paramString1, @Query("user_id") long paramLong, @Query("restrict") String paramString2, @Query("tag") String paramString3);

    @FormUrlEncoded
    @POST("/v1/user/follow/delete")
    Observable<ResponseBody> postUnfollowUser(@Header("Authorization") String paramString, @Field("user_id") long paramLong);

    @FormUrlEncoded
    @POST("/v1/user/follow/add")
    Observable<ResponseBody> postFollowUser(@Header("Authorization") String paramString1, @Field("user_id") long paramLong, @Field("restrict") String paramString2);

    @FormUrlEncoded
    @POST("/v1/illust/bookmark/delete")
    Observable<ResponseBody> postUnlikeIllust(@Header("Authorization") String paramString, @Field("illust_id") long paramLong);

    @FormUrlEncoded
    @POST("/v2/illust/bookmark/add")
    Observable<ResponseBody> postLikeIllust(@Header("Authorization") String paramString1, @Field("illust_id") long paramLong, @Field("restrict") String paramString2, @Field("tags[]") List<String> paramList);

    @GET("/v1/search/autocomplete")
    Observable<PixivResponse> getSearchAutoCompleteKeywords(@Header("Authorization") String paramString1, @Query("word") String paramString2);

    @GET("/v1/trending-tags/illust?filter=for_ios")
    Observable<TrendingtagResponse> getIllustTrendTags(@Header("Authorization") String paramString);

    @GET("/v1/search/illust?filter=for_ios")
    Observable<SearchIllustResponse> getSearchIllust(@Query("word") String paramString1, @Query("sort") String paramString2, @Query("search_target") String paramString3, @Query("bookmark_num") Integer paramInteger, @Query("duration") String paramString4, @Header("Authorization") String paramString5);

    @GET("/v1/search/novel")
    Call<PixivResponse> getSearchNovel(@Header("Authorization") String paramString1, @Query("word") String paramString2, @Query("sort") String paramString3, @Query("search_target") String paramString4, @Query("bookmark_num") Integer paramInteger, @Query("duration") String paramString5);

    @GET("/v1/search/user?filter=for_ios")
    Observable<SearchUserResponse> getSearchUser(@Header("Authorization") String paramString1, @Query("word") String paramString2);

    @GET("/v1/search/popular-preview/illust?filter=for_ios")
    Call<PixivResponse> getPopularPreviewIllust(@Header("Authorization") String paramString1, @Query("word") String paramString2, @Query("search_target") String paramString3, @Query("duration") String paramString4);



    @GET("/v2/illust/follow?restrict=public")
    Call<FollowResponse> getNewTab(@Header("Authorization") String Authorization);

    @GET("/v1/user/follower?filter=for_ios")
    Observable<SearchUserResponse> getUserFollower(@Header("Authorization") String paramString, @Query("user_id") long paramLong);

    @GET("/v1/user/following?filter=for_ios")
    Observable<SearchUserResponse> getUserFollowing(@Header("Authorization") String paramString1, @Query("user_id") long paramLong, @Query("restrict") String paramString2);

    @GET("/v1/illust/recommended?filter=for_ios&include_ranking_label=true")
    Observable<RecommendResponse> getRecommend(@Header("Authorization") String Authorization);

    @GET("/v1/illust/detail?filter=for_ios")
    Observable<IllustDetailResponse> getIllust(@Header("Authorization") String paramString, @Query("illust_id") long paramLong);

    @GET("/v2/illust/related?filter=for_android")
    Observable<RecommendResponse> getIllustRecommended(@Header("Authorization") String paramString, @Query("illust_id") long paramLong);

    @GET("/v1/manga/recommended?filter=for_ios")
    Observable<RecommendResponse> getRecommendedMangaList(@Header("Authorization") String paramString1, @Query("include_ranking_illusts") boolean paramBoolean, @Query("bookmark_illust_ids") String paramString2);
    @GET
    Observable<RecommendResponse> getNext(@Header("Authorization") String paramString1, @Url String paramString2);

    @FormUrlEncoded
    @POST("/v2/user/browsing-history/illust/add")
    Observable<ResponseBody> postAddIllustBrowsingHistory(@Header("Authorization") String paramString, @Field("illust_ids[]") List<Long> paramList);

    @GET
    Observable<SearchUserResponse> getNextUserFollowing(@Url String paramString2);

    @GET
    Observable<UserIllustsResponse> getNextHistory(@Header("Authorization") String Authorization, @Url String paramString2);

    @GET("/v1/illust/related?filter=for_ios")
    Observable<AboutPictureResponse> getAboutPicture(@Query("illust_id") int id);

    @GET("/v1/illust/ranking?filter=for_ios")
    Observable<IllustRankingResponse> getIllustRanking(@Header("Authorization") String paramString1, @Query("mode") String paramString2, @Query("date") String paramString3);

    @GET("/v1/illust/ranking?filter=for_ios")
    Observable<ResponseBody> getIllustRanking1(@Header("Authorization") String paramString1, @Query("mode") String paramString2, @Query("date") String paramString3);

    @GET("/v2/illust/follow")
    Observable<IllustfollowResponse> getFollowIllusts(@Header("Authorization") String paramString1, @Query("restrict") String paramString2);

    @GET("/v1/illust/comments")
    Observable<IllustCommentsResponse> getIllustComments(@Header("Authorization") String paramString, @Query("illust_id") long paramLong);

    @FormUrlEncoded
    @POST("v1/illust/comment/add")
    Observable<ResponseBody> postIllustComment(@Header("Authorization") String paramString1, @Field("illust_id") long illust_id, @Field("comment") String comment, @Field("parent_comment_id") Integer parent_comment_id);

    @FormUrlEncoded
    @POST("/v1/mute/edit")
    Observable<PixivResponse> postMuteSetting(@Header("Authorization") String paramString, @Field("add_user_ids[]") List<Long> paramList1, @Field("delete_user_ids[]") List<Long> paramList2, @Field("add_tags[]") List<String> paramList3, @Field("delete_tags[]") List<String> paramList4);

    @GET("/v1/user/detail?filter=for_ios")
    Observable<UserDetailResponse> getUserDetail(@Header("Authorization") String paramString, @Query("user_id") long id);

    @GET("/v1/user/detail?filter=for_ios")
    Observable<ResponseBody> getUserDetailBody(@Header("Authorization") String paramString, @Query("user_id") long id);

    @GET("/v1/user/illusts?filter=for_ios")
    Observable<UserIllustsResponse> getUserIllusts(@Header("Authorization") String paramString1, @Query("user_id") long paramLong, @Query("type") String paramString2);
    @GET("/v1/user/illusts?filter=for_ios")
    Deferred<UserIllustsResponse> getUserIllustsCor(@Header("Authorization") String paramString1, @Query("user_id") long paramLong, @Query("type") String paramString2);
    @GET
    Observable<UserIllustsResponse> getNextUserIllusts(@Header("Authorization") String paramString1, @Url String paramString2);
    @GET
   Deferred<UserIllustsResponse> getNextUserIllustsCor(@Header("Authorization") String paramString1, @Url String paramString2);
    @GET
    Call<UserIllustsResponse> getNextUserIllustsBlock(@Header("Authorization") String paramString1, @Url String paramString2);

    @GET
    Observable<SearchUserResponse> getNextUser(@Header("Authorization") String paramString1, @Url String paramString2);

    @GET
    Observable<ResponseBody> getImageBitmap(@Url String paramString2);

    void getNexttags();
}

File: app/src/main/java/com/perol/asdpl/pixivez/services/AppApiPixivService.kt
/*
 * MIT License
 *
 * Copyright (c) 2019 Perol_Notsfsssf
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE
 */

package com.perol.asdpl.pixivez.services


import com.perol.asdpl.pixivez.responses.*
import io.reactivex.Observable
import okhttp3.MultipartBody
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.*


interface AppApiPixivService {

    @GET("v1/walkthrough/illusts")
    fun walkthroughIllusts(): Observable<IllustNext>

    //    @FormUrlEncoded
    //    @POST("/v1/mute/edit")
    //    public abstract l<PixivResponse> postMuteSetting(@Header("Authorization") String paramString, @Field("add_user_ids[]") List<Long> paramList1, @Field("delete_user_ids[]") List<Long> paramList2, @Field("add_tags[]") List<String> paramList3, @Field("delete_tags[]") List<String> paramList4);
    @GET("/v2/illust/bookmark/detail")
    fun getLikeIllustDetail(@Header("Authorization") paramString: String, @Query("illust_id") paramLong: Long): Observable<BookMarkDetailResponse>

    @GET("v1/user/bookmark-tags/illust")
    fun getIllustBookmarkTags(@Header("Authorization") paramString1: String, @Query("user_id") paramLong: Long, @Query("restrict") paramString2: String): Observable<BookMarkTagsResponse>

    @GET
    fun getNexttags(@Header("Authorization") paramString1: String, @Url paramString2: String): Observable<BookMarkTagsResponse>

    @GET("/v1/spotlight/articles?filter=for_android")
    fun getPixivisionArticles(@Header("Authorization") paramString1: String, @Query("category") paramString2: String): Observable<SpotlightResponse>

    @GET("/v1/user/recommended?filter=for_android")
    fun getUserRecommended(@Header("Authorization") paramString: String): Observable<SearchUserResponse>

    @GET
    fun getNextPixivisionArticles(@Header("Authorization") paramString1: String, @Url paramString2: String): Observable<SpotlightResponse>

    @Streaming
    @GET
    fun getGIFFile(@Url fileUrl: String): Observable<ResponseBody>

    @Multipart
    @POST("/v1/user/profile/edit")
    fun postUserProfileEdit(@Header("Authorization") paramString: String, @Part paramRequestBody: MultipartBody.Part): Observable<ResponseBody>

    @GET("/v1/ugoira/metadata")
    fun getUgoiraMetadata(@Header("Authorization") paramString: String, @Query("illust_id") paramLong: Long): Observable<UgoiraMetadataResponse>

    @GET("/v1/user/browsing-history/illusts")
    fun getIllustBrowsingHistory(@Header("Authorization") paramString: String): Observable<IllustNext>

    @GET("/v1/user/bookmarks/illust")
    fun getLikeIllust(@Header("Authorization") paramString1: String, @Query("user_id") paramLong: Long, @Query("restrict") paramString2: String, @Query("tag") paramString3: String?): Observable<IllustNext>

    @FormUrlEncoded
    @POST("/v1/user/follow/delete")
    fun postUnfollowUser(@Header("Authorization") paramString: String, @Field("user_id") paramLong: Long): Observable<ResponseBody>

    @FormUrlEncoded
    @POST("/v1/user/follow/add")
    fun postFollowUser(@Header("Authorization") paramString1: String, @Field("user_id") paramLong: Long, @Field("restrict") paramString2: String): Observable<ResponseBody>

    @FormUrlEncoded
    @POST("/v1/illust/bookmark/delete")
    fun postUnlikeIllust(@Header("Authorization") paramString: String, @Field("illust_id") paramLong: Long): Observable<ResponseBody>

    @FormUrlEncoded
    @POST("/v2/illust/bookmark/add")
    fun postLikeIllust(@Header("Authorization") paramString1: String, @Field("illust_id") paramLong: Long, @Field("restrict") paramString2: String, @Field("tags[]") paramList: List<String>?): Observable<ResponseBody>

    @GET("/v2/search/autocomplete?merge_plain_keyword_results=true")
    fun getSearchAutoCompleteKeywords(@Header("Authorization") paramString1: String, @Query("word") paramString2: String?): Observable<PixivResponse>

    @GET("/v1/trending-tags/illust?filter=for_android")
    fun getIllustTrendTags(@Header("Authorization") paramString: String): Observable<TrendingtagResponse>

    @GET("/v1/search/illust?filter=for_android&merge_plain_keyword_results=true")
    fun getSearchIllust(@Query("word") paramString1: String, @Query("sort") paramString2: String, @Query("search_target") paramString3: String?, @Query("bookmark_num") paramInteger: Int?, @Query("duration") paramString4: String?, @Header("Authorization") paramString5: String): Observable<SearchIllustResponse>

    @GET("/v1/search/popular-preview/illust?filter=for_android&merge_plain_keyword_results=true")
    fun getSearchIllustPreview(@Query("word") paramString1: String, @Query("sort") paramString2: String, @Query("search_target") paramString3: String?, @Query("bookmark_num") paramInteger: Int?, @Query("duration") paramString4: String?, @Header("Authorization") paramString5: String): Observable<SearchIllustResponse>

    @GET("/v1/search/novel")
    fun getSearchNovel(@Header("Authorization") paramString1: String, @Query("word") paramString2: String, @Query("sort") paramString3: String, @Query("search_target") paramString4: String, @Query("bookmark_num") paramInteger: Int?, @Query("duration") paramString5: String): Call<PixivResponse>

    @GET("/v1/search/user?filter=for_android")
    fun getSearchUser(@Header("Authorization") paramString1: String, @Query("word") paramString2: String): Observable<SearchUserResponse>

    @GET("/v1/search/popular-preview/illust?filter=for_android")
    fun getPopularPreviewIllust(@Header("Authorization") paramString1: String, @Query("word") paramString2: String, @Query("search_target") paramString3: String, @Query("duration") paramString4: String): Call<PixivResponse>


    @GET("/v1/user/follower?filter=for_android")
    fun getUserFollower(@Header("Authorization") paramString: String, @Query("user_id") paramLong: Long): Observable<SearchUserResponse>

    @GET("/v1/user/following?filter=for_android")
    fun getUserFollowing(@Header("Authorization") paramString1: String, @Query("user_id") paramLong: Long, @Query("restrict") paramString2: String): Observable<SearchUserResponse>

    @GET("/v1/illust/recommended?filter=for_android&include_ranking_label=true")
    fun getRecommend(@Header("Authorization") Authorization: String): Observable<RecommendResponse>

    @GET("/v1/illust/detail?filter=for_android")
    fun getIllust(@Header("Authorization") paramString: String, @Query("illust_id") paramLong: Long): Observable<IllustDetailResponse>

    @GET("/v1/illust/detail?filter=for_android")
    suspend fun getIllustCor(@Header("Authorization") paramString: String, @Query("illust_id") paramLong: Long): IllustDetailResponse

    @GET("/v2/illust/related?filter=for_android")
    fun getIllustRecommended(@Header("Authorization") paramString: String, @Query("illust_id") paramLong: Long): Observable<RecommendResponse>

    @GET("/v2/illust/related?filter=for_android")
    fun getIllustRecommendedNext(@Header("Authorization") paramString: String, @Query("illust_id") paramLong: Long, @Query("seed_illust_ids%5B0%5D") longIdSame: Long, @Query("offset") offset:Int): Observable<RecommendResponse>

    @GET("/v1/manga/recommended?filter=for_android")
    fun getRecommendedMangaList(@Header("Authorization") paramString1: String, @Query("include_ranking_illusts") paramBoolean: Boolean, @Query("bookmark_illust_ids") paramString2: String): Observable<RecommendResponse>

    @GET
    fun getNext(@Header("Authorization") paramString1: String, @Url paramString2: String): Observable<RecommendResponse>

    @FormUrlEncoded
    @POST("/v2/user/browsing-history/illust/add")
    fun postAddIllustBrowsingHistory(@Header("Authorization") paramString: String, @Field("illust_ids[]") paramList: List<Long>): Observable<ResponseBody>

    @GET
    fun getNextUserFollowing(@Url paramString2: String): Observable<SearchUserResponse>

    @GET
    fun getNextHistory(@Header("Authorization") Authorization: String, @Url paramString2: String): Observable<IllustNext>


    @GET("/v1/illust/ranking?filter=for_android")
    fun getIllustRanking(@Header("Authorization") paramString1: String, @Query("mode") paramString2: String, @Query("date") paramString3: String?): Observable<IllustNext>

    @GET("/v1/illust/ranking?filter=for_android")
    fun getIllustRanking1(@Header("Authorization") paramString1: String, @Query("mode") paramString2: String, @Query("date") paramString3: String): Observable<ResponseBody>

    @GET("/v2/illust/follow")
    fun getFollowIllusts(@Header("Authorization") paramString1: String, @Query("restrict") paramString2: String): Observable<IllustNext>

    @GET("/v1/illust/comments")
    fun getIllustComments(@Header("Authorization") paramString: String, @Query("illust_id") paramLong: Long): Observable<IllustCommentsResponse>

    @GET("/v1/illust/comments")
    fun getIllustNextComments(@Header("Authorization") paramString: String, @Url paramString2: String): Observable<IllustCommentsResponse>

    @FormUrlEncoded
    @POST("v1/illust/comment/add")
    fun postIllustComment(@Header("Authorization") paramString1: String, @Field("illust_id") illust_id: Long, @Field("comment") comment: String, @Field("parent_comment_id") parent_comment_id: Int?): Observable<ResponseBody>

    @FormUrlEncoded
    @POST("/v1/mute/edit")
    fun postMuteSetting(@Header("Authorization") paramString: String, @Field("add_user_ids[]") paramList1: List<Long>, @Field("delete_user_ids[]") paramList2: List<Long>, @Field("add_tags[]") paramList3: List<String>, @Field("delete_tags[]") paramList4: List<String>): Observable<PixivResponse>

    @GET("/v1/user/detail?filter=for_android")
    fun getUserDetail(@Header("Authorization") paramString: String, @Query("user_id") id: Long): Observable<UserDetailResponse>

    @GET("/v1/user/detail?filter=for_android")
    fun getUserDetailBody(@Header("Authorization") paramString: String, @Query("user_id") id: Long): Observable<ResponseBody>

    @GET("/v1/user/illusts?filter=for_android")
    fun getUserIllusts(@Header("Authorization") paramString1: String, @Query("user_id") paramLong: Long, @Query("type") paramString2: String): Observable<IllustNext>

    @GET("/v1/user/illusts?filter=for_android")
    suspend fun getUserIllustsCor(@Header("Authorization") paramString1: String, @Query("user_id") paramLong: Long, @Query("type") paramString2: String): IllustNext

    @GET
    fun getNextUserIllusts(@Header("Authorization") paramString1: String, @Url paramString2: String): Observable<IllustNext>

    @GET
    suspend fun getNextUserIllustsCor(@Header("Authorization") paramString1: String, @Url paramString2: String): IllustNext

    @GET
    fun getNextUserIllustsBlock(@Header("Authorization") paramString1: String, @Url paramString2: String): Call<IllustNext>

    @GET
    fun getNextUser(@Header("Authorization") paramString1: String, @Url paramString2: String): Observable<SearchUserResponse>

    @GET
    fun getImageBitmap(@Url paramString2: String): Observable<ResponseBody>

}
